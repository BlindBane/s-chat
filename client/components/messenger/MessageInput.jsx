import React, { Component } from 'react'

class MessageInput extends Component {
  state = {
    messageText: '',
    channel: 'general'
  }
  handleChange = (e) => {
    e.preventDefault()
    const messageText = e.target.value
    const lastCharIndex = messageText.length - 1
    if (messageText[0] === '/' && messageText[lastCharIndex] === ' ') {
      this.setState({
        messageText: '',
        channel: messageText.slice(1, lastCharIndex)
      })
    } else {
      this.setState({
        messageText: e.target.value
      })
    }
  }
  handleSubmit = (e) => {
    e.preventDefault()
    const messageText = this.state.messageText
    const lastCharIndex = messageText.length
    if (messageText[0] === '/') {
      this.setState({
        messageText: '',
        channel: messageText.slice(1, lastCharIndex)
      })
    } else {
      const newMsg = {
        userId: Math.random(),
        username: 'Bill',
        messageText: this.state.messageText,
        channel: this.state.channel
      }
      this.props.handleNewMsg(newMsg)
      this.setState({
        messageText: ''
      })
    }
  }
  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          <span>[Tell {this.state.channel}: ] - </span>
          <input
            type='text'
            value={this.state.messageText}
            onChange={this.handleChange}
          />
        </label>
        <input type='submit' value='Submit' />
      </form>
    )
  }
}

export default MessageInput
