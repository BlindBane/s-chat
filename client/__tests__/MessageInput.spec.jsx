import React from 'react'
import { shallow } from 'enzyme'
import MessageInput from './../components/messenger/MessageInput'

test('MessageInput renders correctly', () => {
  const component = shallow(<MessageInput />)
  expect(component).toMatchSnapshot()
})
