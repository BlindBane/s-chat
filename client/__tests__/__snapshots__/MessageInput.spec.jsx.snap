// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    <span>
      [Tell 
      general
      : ] - 
    </span>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
